### PR TITLE
Don't create non-existent queue and dead letter queue.

### LINKED ISSUE #109: Remove creation of non-existing queue, dead letter queue and exchanges.

### PR DESCRIPTION
**Pull request by yojagad**
Fixes #109

### CODE REVIEW

**Review by pragnagopa — CHANGES_REQUESTED** [review]
- Please create an issue or link existing issue for this PR
- I believe we should support moving messages to deadletterexchange queue as long as user specifies it. PR should only remove creating queue / dead letter exchange but should allow using existing queues

**Comment by yojagad** [comment]
@pragnagopa Correct me if I'm wrong, but the only action that moves messages to the deadletter exchange and the queue is https://github.com/Azure/azure-functions-rabbitmq-extension/blob/dev/src/Trigger/RabbitMQListener.cs#L178 . As long as the user has them configured correctly and bound the exchange to the queue, that line would either move it to dead letter exchange or they might end up losing the message altogether.

I specifically didn't attempt to do a passive check on whether the exchange or the dead letter queue exists, because there's no reason that the binding itself should fail to come up if either of these aren't configured correctly. Let me know if I'm missing something else though, please.

File: src/Services/RabbitMQService.cs [hunk 1]
```diff
@@ -33,29 +31,13 @@ public RabbitMQService(string connectionString, string hostName, string userName
...
             _rabbitMQModel = new RabbitMQModel(_model);
-
-            _deadLetterExchangeName = deadLetterExchangeName;
             _queueName = queueName ?? throw new ArgumentNullException(nameof(queueName));
 
-            Dictionary<string, object> args = new Dictionary<string, object>();
-
-            // Create dead letter queue
-            if (!string.IsNullOrEmpty(_deadLetterExchangeName))
-            {
```
Comment (pragnagopa) [inline src/Services/RabbitMQService.cs:48]: We should allow users to specify a `deadLetterExchangeName` . I think you just have to delete creating the queue here. Also,  log an error if adding message to `deadLetterExchangeName` fails

File: src/RabbitMQAttribute.cs [hunk 2]
```diff
@@ -40,8 +39,5 @@ public sealed class RabbitMQAttribute : Attribute
 
         [ConnectionString]
         public string ConnectionStringSetting { get; set; }
-
-        [AutoResolve]
-        public string DeadLetterExchangeName { get; set; }
```
Comment (pragnagopa) [inline src/RabbitMQAttribute.cs:45]: Without this users will not be able to specify an existing queue for `DeadLetterExchangeName` ?

File: README.md [hunk 3]
```diff
@@ -36,7 +36,6 @@ The above sample waits on a trigger from the queue named "queue" connected to th
 |--|--|--|
 |ConnectionStringSetting|The connection string for the RabbitMQ queue|`amqp://user:password@url:port`|
 |QueueName|The name of the source or destination queue|`myQueue`|
```
Comment (pragnagopa) [inline README.md]: nit:  Add `To move failed messages to deadletterqueue, configure queue (add link)`

**Review by pragnagopa — APPROVED** [review]
Please run all the samples before merging the PR

**Comment by yojagad** [comment]
Ran through the samples and didn't find issues.

**Pull request merged**